datafusion/sql/src/unparser/dialect.rs

```diff
@@ -231,6 +319,19 @@ impl Dialect for SqliteDialect {
     fn supports_column_alias_in_table_alias(&self) -> bool {
         false
     }
+
+    fn scalar_function_to_sql_overrides(
+        &self,
+        unparser: &Unparser,
+        func_name: &str,
+        args: &[Expr],
+    ) -> Result<Option<ast::Expr>> {
+        if func_name == "date_part" {
```

Comment:
`if` based approach won't scale - we need a hashmap / registry IMO for this

Reply:
or a match statement that delegates to different functions. That would probably be faster than a hashmap if we're not planning to make this dynamic